lib/we/call/connection.rb

```diff
@@ -100,14 +100,19 @@ def setup_sunset_middleware(faraday)
       # @return [String] Environment (usually 'development', 'staging', 'production', etc.)
       def guess_env
         return config.app_env if config.app_env
-        ENV['RAILS_ENV'] || ENV['RACK_ENV']
+        ENV['RACK_ENV'] || rails_app_env
       end
 
       # @return [String] Check for config.app_name, or detect name from Rails application
       def guess_app
         return config.app_name if config.app_name
-        return ENV['APP_NAME'] if ENV['APP_NAME']
-        rails_app_name
+        ENV['APP_NAME'] || rails_app_name
+      end
+
+      def rails_app_env
+        if (defined? ::Rails) && !::Rails.env.nil?
```

Comment:
the second check is redundant since you'll be returning nil if it doesn't pass either of those.

Reply:
You'd be surprised. There are some situations (where Rails is kinda loaded but not instantiated) where the Rails class might exist but the module-level variable has not been set.